src/components/TodoList.jsx

```diff
@@ -1,64 +1,158 @@
 import React from 'react';
 import TodoPoint from './TodoPoint';
+import formatDate from './utils';
 
 const todoStyle = {
     display: "inline-block"
 }
-
-
+// Sort users by several categories
+const filterTasks = (sortingOpt) => {
+    {
+        switch (sortingOpt) {
+            // Sort by name ascending 
+            case "Name asc":
+                return (task1, task2) => (task1.taskName.toLowerCase() > task2.taskName.toLowerCase()) ? 1 : -1
+            // Sort by name descending 
+            case "Name desc":
+                return (task1, task2) => (task1.taskName.toLowerCase() < task2.taskName.toLowerCase()) ? 1 : -1
+            // Sort by newer tasks
+            case "Newer":
+                return ((task1, task2) => (task2.date - task1.date))
+            // Sort by older tasks
+            case "Older":
+                return (task1, task2) => (task1.date - task2.date)
+            default:
+                console.log("'Sort by' opt set or somthg went wrong")
+        }
+    }
+}
 
 class TodoListXP extends React.Component {
     constructor(props) {
         super(props);
-
         this.state = {
             tasks: [
-                "do one thing",
-                "another one",
-                "last one to do"
+                { taskName: "do one thing", date: new Date() },
+                { taskName: "another one", date: new Date() },
+                { taskName: "last one to do", date: new Date() }
             ],
-            input: "nichego net"
+            input: "",
+            sortingOpt: "Sort by",
```

Comment:
better be precise with naming. Don't save on symbols for no reason: `sortingOption`